lang/c/src/st.c

```diff
@@ -212,7 +212,7 @@ void st_free_table(st_table *table)
 }
 
 #define PTR_NOT_EQUAL(table, ptr, hash_val, key) \
-((ptr) != 0 && (ptr->hash != (hash_val) || !EQUAL((table), (key), (ptr)->key)))
+((ptr) != 0 && (ptr->hash != (hash_val) || !EQUAL((table), (void*) (key), (void*) (ptr)->key)))
```

Comment:
Are the explicit casts to `void*` needed here and below for MSVC? Or is there some other reason to have them?

Reply:
without the `void*` casts, we get `-Wpointer-integer-compare` warnings due to `st_data_t` being a `uintptr_t`

the other call-sites throughout `/avro/lang/c` pass in pointers so the cast is only needed for `st.c`